### PR TITLE
Restructure site to allow easy translations

### PR DESCRIPTION
**Pull request by GjjvdBurg**
Hi there!

First off, Precious Plastic seems like a really cool project!

This pull request is inspired by issue #59. I figured it would be really useful if the site could be translated, to make it more available for people in non-English speaking countries. I've used this [article](https://developmentseed.org/blog/multilingual-jekyll-sites/) and this [repo](https://github.com/tristen/multilingual) as sources for the changes.

As you can see, the site structure has been changed quite a bit. I've changed the following:
- The main `index.html` is now a redirect to the English site, which is moved to `en` directory.
- Each page has a `lang` variable in the frontmatter for the language abbreviation.
- The permalink in the frontmatter of the pages has been removed, since it's not needed anymore with the new site structure.
- All internal links on the site now come with a `{{page.lang}}` prefix, to redirect to the right language page.
- Language-specific text on the pages has been replaced by a lookup in a translations variable, which is defined in the main layout `default.html`. This way, there are no English terms hard-coded into the pages anymore. Instead, they are looked up depending on the language of the current page.
- This translations variable in the layout can be used to translate both the filename of the html file, as well as the page title. For example, for a Dutch translation the "about.html" page could be named "over_ons.html", as long as in the "about" field of the translations, we accordingly have "link: over_ons". This way, even the URL of the page can make sense in the specific language (see the note below on the exception with the "video" directory). See [this commit](https://github.com/hakkens/precious-plastic/blob/85d135b741d709bb8dbfb87a101393a74022b82c/_layouts/default.html) for an example.

To translate the website, a contributor could now do the following:
1. Copy the `en` folder to a new folder, for example `nl`.
2. Translate all the English text in the pages to the desired text, and change the `lang` variable in the frontmatter to `nl`.
3. Update the translations in the `_layouts/default.html` file, by copying the entire `en` section, to a section with title `nl`, and translate all the `title` fields, and where necessary, the `link` field.
4. Add the language definition to `_config.yml`, with the name of the language given in that language itself. So "Nederlands" instead of "Dutch", "Deutsch" instead of "German", etc. (See below for more on this point).
5. Finally, submit a pull request with these additions.
   (And if it's a really awesome contributor, he/she could even help out with subtitles of the videos of course!)

What I haven't done:
- Create a language selection dropdown somewhere. I figured this would be something you might like to think about yourself if you decide to actually implement this. This would use the language definitions in the `_config.yml` file. See the sources mentioned above for ideas on how to do this.
- Allow the "videos" directory to be renamed. This can be done, but since "videos" is hardcoded in quite a lot of places, it would be more hassle than it seems worth at this point.
- Do something with the assets, such as images, that contain English text. Ideally these would also be translated of course.

Hope this helps! Let me know if you need more info, or if there's anything else I can help with.

### CODE REVIEW

**Comment by leo** [comment]
@GjjvdBurg First of all: Thank you very much for your contribution! I've looked into it and it seems to be a pretty great solution for our language problem. I have a few concerns though:
- Is it possible that you could use [Front Matter defaults](https://jekyllrb.com/docs/configuration/#front-matter-defaults) so that pages/files within the "en" directory automatically get the "lang" property set to "en"? That would be awesome! This way, contributors won't have to change the "lang" propery on each of their pages, but rather just add it once in the `config.yml`.
- Obviously, we would have to adjust all of the copies of a certain page if we want to make changes to the code. But I guess if we really want to support multiple languages, we'll have to do this step...
- @davehakkens, @mattia-io and I only speak three different languages combined: German, Italian (I guess) and Dutch. So all in all, we probably won't be able to except contributions in any other language, since we're not able to control what exactly has been added to the site. But I'm sure that we could find a way to solve this by simply using services like [Gengo](https://gengo.com).

I will dive a little deeper into this whole thing later today and then I'll let you know if we can accept it. But currently, it seems like an awesome idea! :blush:

File: _includes/header.html [hunk 1]
```diff
@@ -15,6 +15,10 @@
 
       {% for node in sorted %}
 
+				{% if node.lang != page.lang %}
```
Comment (leo) [inline _includes/header.html]: Seems like the indentation broke here. Can you fix this? :blush:

File: index.html [hunk 2]
```diff
@@ -1,88 +1,9 @@
...
-      <a href="#">Spread!</a>
-    </div>
-
-    <div class="visitwebsite"><a href="/plan">visit website</a></div>
-  </article>
-</section>
+<!DOCTYPE html>
+<html>
+	<head>
+		<meta http-equiv='content-type' content='text/html; charset=utf-8' />
```
Comment (leo) [inline index.html]: Can you please replace the apostrophes with quotation marks (`"`)? It also looks like the indentation is set to tabs (or too many spaces), would be great if you could convert them to 2 spaces... :blush:

**Comment by thisismattia** [comment]
@GjjvdBurg great news!
Thanks for chipping in your time and skills! Much appreciated from all of us.

Sounds very exciting, we could use the service Leo mentioned or even a simple gDrive shared document for each language where we can invite (or block if they needed) people to translate, comment and discuss possible translations for the website. Maybe a stupid idea, don't know. 

But all this sounds amazing, We have tons of people willing to help with this.

*I also speak Spanish. And I have many trusted friends speaking chinese and japanise if that was needed. Also French won't be a problem.

Good day!

**Comment by leo** [comment]
> I also speak Spanish. And I have many trusted friends speaking chinese and japanise if that was needed. Also French won't be a problem.

👌 Spanish, Chinese, Japanese, French, Dutch and German. Sounds like a great start!

**Comment by GjjvdBurg** [comment]
Hi everyone,

Glad you like it! I've made the changes that @leo suggested, as you can see. In terms of translations, you could indeed start with trusted friends, and if other people want to help you could think of a scheme where you'd wait for someone to review a pull request with a translation, before accepting it. Anyway, if there's anything else I can help with let me know!

**Comment by leo** [comment]
@GjjvdBurg Awesome! 👌Currently, there only seems to be a single thing left to do: You need to pull the latest commits from the repo and solve the conflicts. After that, I will be able to merge your PR.

**Comment by GjjvdBurg** [comment]
@leo I may have messed it up. I had renamed the `gh-pages` branch to `master` in my fork, to avoid multiple versions of the site to be online simultaneously. But now I can't seem to pull the changes of your branch and rebase the `dev` branch on it. I think the easiest and _cleanest_ way would be to delete my fork, make a new one, and apply the changes. Agreed?

**Comment by leo** [comment]
@GjjvdBurg Oh... Yep, seems like a good idea! 😊 No need to copy over the description from here, by the way.

**Comment by GjjvdBurg** [comment]
@leo Alright, I'm on it.

**Pull request closed without merging**